### PR TITLE
Fix missing unread badge

### LINKED ISSUE #893: No longer getting unread badge
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/caprine/issues/893)

Looks like the messenger.com site updated last night to the newer UI.  As of this morning (unsure if it is related to the update or not) I am no longer getting the unread badge.  I have verified that the option in Caprine to show the unread badge is turned on.

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/caprine/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### LINKED ISSUE #905: Not getting notified after latest update
Hello, i updated caprine to the latest version (2.31.0)

and now i don't get notified when i receive a new message, i'm on linux mint 19.1

### PR DESCRIPTION
**Pull request by pkuczynski**
Fixes #893
Fixes #905

### CODE REVIEW

**Comment by pkuczynski** [comment]
There is also a reference to `._7t0d` in browser.css and I am not sure if this should be updated? It's marked as `Hide profile picture`, which I am not sure why anybody would like to have enabled?

**Comment by sindresorhus** [comment]
> There is also a reference to ._7t0d in browser.css and I am not sure if this should be updated? It's marked as Hide profile picture, which I am not sure why anybody would like to have enabled?

That's unrelated to this.

File: source/browser.ts [hunk 1]
```diff
@@ -416,111 +416,109 @@ function closePreferences(): void {
...
-// 	});
-// }
+async function getDataUrlFromImg(img: HTMLImageElement, unread: boolean): Promise<string> {
+	// eslint-disable-next-line no-async-promise-executor
+	return new Promise(async resolve => {
+		if (unread) {
+			const dataUnreadUrl = img.getAttribute('dataUnreadUrl');
+
+			if (dataUnreadUrl) {
+				return resolve(dataUnreadUrl);
```
Comment (sindresorhus) [inline source/browser.ts]: This creates misleading code as it makes it look like `resolve` actually returns something.
```suggestion
				resolve(dataUnreadUrl);
				return;
```

Comment (pkuczynski) [reply source/browser.ts]: I actually just reverted this piece of code from the previous version, but sure, I can make this change!

Comment (pkuczynski) [reply source/browser.ts]: Done!

**Comment by sindresorhus** [comment]
I tried out this PR now, but I'm not getting any unread badge for new messages on macOS.

**Comment by pkuczynski** [comment]
I do. So maybe it's just a matter of different version being served for my account (A/B testing). The only thing I changed from the previous implementation are selectors - in particular https://github.com/sindresorhus/caprine/pull/908/files#diff-2c24973e0432bc28df516c6328533f5dR424

Can you check what would be the selector for pre profile picture for you?

Screenshot 2019-05-27 at 18 09 40

**Comment by pkuczynski** [comment]
I think I found another issue with group chats. Working on this... Stay tuned.

**Comment by pkuczynski** [comment]
@sindresorhus I have an improved fix, which:

* moves all conversation list logic run in browser into a separate file
* parse conversation list more based on aria or data attributes, and less on styles, so it should be much more error resistant
* introduce error handling for loading conversation icons (lack of this was causing missing badge icon)
* adds some sane fallbacks for the future when the new messenger is released with changed structure

Looking forward for your feedback and getting this released soon. I have no new message notification in a current version which drives me nuts :)

**Comment by pkuczynski** [comment]
@sindresorhus did you maybe had a chance to look into this? Apologies for being annoying, but you would not imagine how much it drives me nuts to not know when someone sent me a message and having to switch to Caprine all the time... :/

**Comment by sindresorhus** [comment]
Thank you for fixing this! 🙌

It's been driving me crazy too.

**Comment by pkuczynski** [comment]
Your welcome. Thanks for the great job you did so far! :)

**Pull request merged**